lib/browser/guest-window-manager.ts

```diff
@@ -59,6 +60,23 @@ export function openGuestWindow ({ embedder, guest, referrer, disposition, postD
     }
   }
 
+  if (createWindow) {
+    const webContents = createWindow({
+      webContents: guest,
+      ...browserWindowOptions
+    });
+
+    webContents.loadURL(url, {
+      httpReferrer: referrer,
+      ...(postData && {
+        postData,
+        extraHeaders: formatPostDataHeaders(postData as Electron.UploadRawData[])
+      })
+    });
```

Comment:
the code below only does this `if (!guest)`, we should do that here too.

I think we also want to do `handleWindowLifecycleEvents`, as that's what makes `outlivesOpener` work correctly. That method will have to be refactored to take a WebContents instead of a BrowserWindow, but I think that is better actually: use the `destroyed` event of the `WebContents` instead of the `closed` event of `BrowserWindow`.

We can't emit `did-create-window` since that event takes a BrowserWindow as its first argument, but I think that's okay, since it would happen at the same time that the `createWindow` function is called anyway, so it wouldn't provide any additional capability. We should probably document that that event won't be emitted if you specify `createWindow` though.